File: 169.多数元素.py
#
# @lc app=leetcode.cn id=169 lang=python3
#
# [169] 多数元素
#

# @lc code=start
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        result = {}
        res = 0
        maxSize = 0

        for i in range(len(nums)) :
            if nums[i] not in result :
                result[nums[i]] = 1
            else :
                result[nums[i]] += 1
            
        for key, value in result.items() :
            if value > maxSize :
                maxSize = value
                res = key
        
        return res
# @lc code=end

File: 169.多数元素.cpp
/*
 * @lc app=leetcode.cn id=169 lang=cpp
 *
 * [169] 多数元素
 */

// @lc code=start
#include <iostream>
using namespace std;
#include <vector>
#include <map>
class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        map<int, int> result;
        int res = 0;
        int maxSize = 0;

        for(int i = 0; i < nums.size(); i++)
        {
            if(result.find(nums[i]) == result.end())
            {
                result.insert(pair<int, int>(nums[i], 1));
            }
            else
            {
                result[nums[i]]++;
            }
        }


        for(map<int, int>::iterator it = result.begin(); it != result.end(); it++)
        {
            if(it->second > maxSize)
            {
                maxSize = it->second;
                res = it->first;
            }
        }

        if(maxSize > nums.size() / 2)
        {
            return res;
        }
        else
        {
            return 0;
        }
    }
};
// @lc code=end
